lib/rate-limit.js

```diff
@@ -0,0 +1,85 @@
+var Promise = require('promise'),
+    debug = require('debug')('pulldasher:rate-limit');
```

Comment:
there might be a module for this but this is short and concise so im cool with this